modules/headline.py

```diff
@@ -4,52 +4,51 @@
-			article.parse()
-			return article
-		except newspaper.article.ArticleException:  # List possible errors in case of any exception
-			print("\nCONNECTION/URL ERROR: Article could not be retrieved.")
-
-	# Adding Training/Testing Data
-	def train(self, headline):
-		with open("positive_headlines.csv") as file:
-			for sentence in file:
-				self.pos.append([{word: True for word in nltk.word_tokenize(sentence)}, 'Positive'])
-
-		with open("negative_headlines.csv") as file:
-			for sentence in file:
-				self.neg.append([{word: True for word in nltk.word_tokenize(sentence)}, 'Negative'])
-
-		training = self.pos[:int(len(self.pos))] + self.neg[:int(len(self.neg))]
-
-		classifier = NaiveBayesClassifier.train(training)  # Training
-		sentiment = classifier.classify({word: True for word in nltk.word_tokenize(headline)})
-		return sentiment
-
-	# categorize headline
-	def headline_category(self, headline, sentiment):
-		print("\nHEADLINE  :", headline.upper())
-		print("SENTIMENT :", sentiment)
-		print("AUTHOR(S) :", self.extract_headline().authors, '\n')
-
-	# main of class
-	def main(self):
-		hdln = self.extract_headline().title.strip()
-		sntmnt = self.train(hdln)
-		self.train(hdln)
-		self.headline_category(hdln, sntmnt)
-		
-		
+    # Initialisations
+    def __init__(self):
+        self.news_url = input("\nEnter The URL : ")
+        self.pos = []  # Variable to store all positive tokens from positive_headlines.csv file
+        self.neg = []  # Variable to store all negative tokens from negative_headlines.csv file
+        self.article = newspaper.Article(self.news_url)
+
+    # extract headline
+    def extract_article(self):
+        try:
+            self.article.download()
+            self.article.parse()
+            return self.article
+        except newspaper.article.ArticleException:  # List possible errors in case of any exception
+            print("\nCONNECTION/URL ERROR: Article could not be retrieved.")
+
+    # Adding Training/Testing Data
+    def train(self, headline):
+        with open("positive_headlines.csv") as file:
+            for sentence in file:
+                self.pos.append([{word: True for word in nltk.word_tokenize(sentence)}, 'Positive'])
+
+        with open("negative_headlines.csv") as file:
+            for sentence in file:
+                self.neg.append([{word: True for word in nltk.word_tokenize(sentence)}, 'Negative'])
+
+        training = self.pos[:int(len(self.pos))] + self.neg[:int(len(self.neg))]
+
+        classifier = NaiveBayesClassifier.train(training)  # Training
+        sentiment = classifier.classify({word: True for word in nltk.word_tokenize(headline)})
+        return sentiment
+
+    def headline_category(self, headline, sentiment):
+        print("\nHEADLINE  :", headline.upper())
+        print("SENTIMENT :", sentiment)
+        print("AUTHOR(S) :", self.article.authors, '\n')
+
+    # main of class
+    def main(self):
+        article = self.extract_article()
```

Comment:
Why would you define article here if self.article is already initiated?
There are 2 ways around this;
1. Instead of `article = self.extract_article()` say `self.extract_article()`
2. In `def __init__` make `self.article = self.extract_article()` where the entired article is initiated, downloaded and parsed. Here you would want the return.

Reply:
I prefer 2